### PR TITLE
test: Add an integration test for DNS requests

### PR DESCRIPTION
**Pull request by vadorovsky**
Add test cases for DNS requests which use dns-mock-server[0], look for the related network events and try to parse them.

Move `parse_dns` function out of `mod pulsar`, so it can be used in tests.

### CODE REVIEW

File: crates/modules/network-monitor/src/lib.rs [hunk 1]
```diff
@@ -648,4 +664,83 @@ pub mod test_suite {
...
+                        return false;
+                    }
+                    let data = data.bytes(&event.buffer).unwrap();
+                    match parse_dns(data) {
+                        Some(Payload::DnsResponse { questions, answers }) => {
+                            questions.iter().any(|q| q.name == "example.io")
+                                && answers.iter().any(|a| {
+                                    a.name == "example.io"
+                                        && (a.data == "A(Record(127.0.0.1))"
+                                            || a.data == "AAAA(Record(::1))")
```
Comment (banditopazzo) [inline crates/modules/network-monitor/src/lib.rs:735]: the record here is hardcoded, the test only works for localhost

**Review by banditopazzo — CHANGES_REQUESTED** [review]

File: crates/modules/network-monitor/src/lib.rs (hunk 1 above)
Comment (vadorovsky) [reply crates/modules/network-monitor/src/lib.rs:735]: Well, I added the "example.io -> localhost" (both v4 and v6 representation of localhost) entry to the mock DNS and I'm asserting that exact event only. So I don't really see the problem.

Comment (vadorovsky) [reply crates/modules/network-monitor/src/lib.rs:735]: But I can add more domains and addresses if that's what you mean.

Comment (vadorovsky) [reply crates/modules/network-monitor/src/lib.rs]: Gotcha. Should be resolved now.

**Review by banditopazzo — APPROVED** [review]

**Pull request merged**